service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -63,13 +73,47 @@ public RunSetsApiController(
 
   @Override
   public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
-    if (!requestIsValid(request, this.cbasApiConfiguration)) {
-      return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+    Gson gson = new Gson();
+
+    // request validation
+    Optional<ResponseEntity<RunSetStateResponse>> validateRequestResponse =
+        validateRequest(request, this.cbasApiConfiguration.getRunSetsMaximumRecordIds());
+    if (validateRequestResponse.isPresent()) {
+      return validateRequestResponse.get();
     }
 
-    // Store the method
-    UUID methodId = UUID.randomUUID();
+    // Fetch WDS Records and keep track of errors while retrieving records
+    String recordType = request.getWdsRecords().getRecordType();
+    List<String> recordIds = request.getWdsRecords().getRecordIds();
+    recordRecordsInRequest(recordIds.size());
 
+    ArrayList<RecordResponse> recordResponses = new ArrayList<>();
+    HashMap<String, String> recordIdsWithError = new HashMap<>();
```

Comment:
Nit: This is more a list of pairs than a map. Or... I also wonder whether reversing the key/value order would let us combine similarly-erroring runs together?

Reply:
Yeah it can be a list of pairs. I guess I found Map fitting my mental model as each error would be associated with 1 Record Id. 
>I also wonder whether reversing the key/value order would let us combine similarly-erroring runs together

WDS returns an `ErrorResponse` containing the below fields and hence putting that as key to combine similarly-erroring runs might not work because of timestamp field 🤔 
```
 {"timestamp":"2022-10-20T19:45:01.210+00:00","status":404,"error":"Not Found","message":"Instance does not exist","path":"/15f36863-30a5-4cab-91f7-52be439f1175/records/v0.2/FOO/FOO4"}
```